File: CourseProjects/CourseProjects/VisualStudioProjects/Languages/Basic VB Programming/Interfaces/Triangle.vb
Public Class Triangle
    Inherits Shape
    Public Overrides Sub paint(ByVal color As Integer)
    End Sub

    Public Overrides Sub draw(ByVal size As Integer)
        For i As Integer = 0 To size
            printLeftSize(size, i)
            printMiddle(size, i)
            printRightSize()
        Next i
        printBase(size)
    End Sub
    Private Sub printLeftSize(ByVal size As Integer, ByVal rowNum As Integer)
        For i As Integer = (size - rowNum) To 1 Step -1
            Console.Write(" ")
        Next i
        Console.Write("/")
    End Sub
    Private Sub printMiddle(ByVal size As Integer, ByVal rowNum As Integer)
        For i As Integer = 0 To rowNum
            Console.Write("  ")
        Next i
    End Sub
    Private Sub printRightSize()
        Console.WriteLine("\")
    End Sub
    Private Sub printBase(ByVal size As Integer)
        Dim baseSize As Integer = size * 2
        Console.Write(" ")
        For i As Integer = 0 To baseSize
            Console.Write("-")
        Next i
        Console.WriteLine()
    End Sub
End Class

File: CourseProjects/CourseProjects/VisualStudioProjects/Languages/Basic CS  Programming/Shapes/Triangle.cs
using System;

namespace Interfaces
{
	public class Triangle : Shape {
		public override void paint(int color) {}
		public override void draw(int size) {
			for(int i=0;i<size;i++) {
				printLeftSize(size,i);
				printMiddle(size,i);
				printRightSize();
			}
			printBase(size);
		}
		private void printLeftSize(int size,int rowNum) {
			for(int x=(size - rowNum);x>0;x--) {
				Console.Write(" ");
			}
			Console.Write("/");
		}
		private void printMiddle(int size,int rowNum) {
			for(int x=0;x<rowNum;x++) {
				Console.Write("  ");
			}
		}
		private void printRightSize() {
			Console.WriteLine(@"\");
		}
		private void printBase(int size) {
			int baseSize = size * 2;
			Console.Write(" ");
			for(int i=0;i<baseSize;i++) {
				Console.Write("-");
			}
			Console.WriteLine();
		}
	}
}
